server/src/test/java/org/cloudfoundry/identity/uaa/scim/endpoints/UserIdConversionEndpointsTests.java

```diff
@@ -222,16 +281,93 @@ public void testBadFilter11() {
 
     @Test
     public void testDisabled() {
-        endpoints = new UserIdConversionEndpoints(provisioning, mockSecurityContextAccessor, scimUserEndpoints, false);
+        arrangeCurrentIdentityZone("uaa");
+        endpoints = new UserIdConversionEndpoints(mockSecurityContextAccessor, scimUserEndpoints, scimUserProvisioning, identityZoneManager, false);
         ResponseEntity<Object> response = endpoints.findUsers("id eq \"foo\"", "ascending", 0, 100, false);
         Assert.assertEquals(HttpStatus.BAD_REQUEST, response.getStatusCode());
         Assert.assertEquals("Illegal Operation: Endpoint not enabled.", response.getBody());
     }
 
     @Test
     public void noActiveIdps_ReturnsEmptyResources() {
-        when(provisioning.retrieveActive(anyString())).thenReturn(Collections.emptyList());
+        arrangeCurrentIdentityZone("uaa");
         SearchResults<?> searchResults = (SearchResults<?>) endpoints.findUsers("username eq \"foo\"", "ascending", 0, 100, false).getBody();
         assertTrue(searchResults.getResources().isEmpty());
     }
+
+    private void arrangeCurrentIdentityZone(final String idzId) {
+        final IdentityZone identityZone = new IdentityZone();
+        identityZone.setId(idzId);
+        idzHolderMockedStatic.when(IdentityZoneHolder::get).thenReturn(identityZone);
+        idzHolderMockedStatic.when(IdentityZoneHolder::getCurrentZoneId).thenReturn(idzId);
+        when(identityZoneManager.getCurrentIdentityZoneId()).thenReturn(idzId);
+    }
+
+    private void arrangeScimUsersForFilter(
```

Comment:
This method is really complex. Is it really necessary to validate all of these details for all of the tests that call this?

Reply:
I assume that you were referring to the method below, i.e., `assertEndpointReturnsCorrectResult`. I simplified it a bit in commit d0cf81f.